### PR TITLE
Add new test application generator for unsubmitted with complete references

### PR DESCRIPTION
**Pull request by benswannack**
## Context
This will be required for the feature in sandbox to prefill application forms for candidates

## Changes proposed in this pull request
Add a new test application generator for unsubmitted with complete references, alongside the existing unsubmitted state

## Link to Trello card
https://trello.com/c/QiEcDyYo/2343-%F0%9F%92%94-decoupled-references-update-test-applications-to-reflect-the-new-references-flow

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by benswannack** [comment]
I set the base of this to not-master so the diff makes more sense - will merge this into master once the base branch is merged

**Comment by mnacos** [comment]
Looking good. You'll probably want to include an `:unsubmitted_with_completed_references` application within https://github.com/DFE-Digital/apply-for-teacher-training/blob/improve-test-applications/app/workers/generate_test_applications.rb#L18

File: spec/lib/test_applications_spec.rb
Comment (duncanjbrown) [inline spec/lib/test_applications_spec.rb:117]: Maybe check that all the references are `feedback_provided` rather than not `feedback_requested`? Just in case they've got into a weird intermediate state

File: spec/lib/test_applications_spec.rb
Comment (duncanjbrown) [inline spec/lib/test_applications_spec.rb:137]: Do we expect these to be `feedback_requested`?

**Comment by tijmenb** [comment]
This needs a rebase again after now that https://github.com/DFE-Digital/apply-for-teacher-training/pull/3190 is merged.

**Review by duncanjbrown — APPROVED** [review]
Great stuff!

**Review by tvararu — APPROVED** [review]

**Pull request merged**
